plugins/modules/nd_compliance_requirement_communication.py

```diff
@@ -0,0 +1,560 @@
+  state:
+    description:
+    - Use C(query) for retrieving the version object.
+    type: str
+    choices: [ query, absent, present ]
+    default: query
+extends_documentation_fragment: cisco.nd.modules
+"""
+
+EXAMPLES = r"""
+- name: Get all compliance communication type requirements
+  cisco.nd.nd_compliance_requirement_communication:
+    insights_group: igName
+    state: query
+  register: query_results
+- name: Get a specific compliance communication type requirement
+  cisco.nd.nd_compliance_requirement_communication:
+    insights_group: igName
+    name: complianceRequirementName
+    state: query
+  register: query_results
+- name: Create compliance communication type requirement
+  cisco.nd.nd_compliance_requirement_communication:
+    insights_group: igName
+    name: complianceRequirementName
+    sites:
+    - siteName1
+    - siteName2
+    enabled: false
+    type: may
+    from_object:
+      type: epg
+      includes:
+        - type: vrf
+          attribute: DN
+          patterns:
+            - type: tenant
+              operator: begins_with
+              value: foo
+            - type: vrf
+              operator: contains
+              value: bar
+        - type: epg
+          attribute: DN
+          patterns:
+            - type: tenant
+              operator: contains
+              value: foo
+            - type: ap
+              operator: contains
+              value: bar
+            - type: epg
+              operator: contains
+              value: foobar
+    to_object:
+      type: epg
+      excludes:
+        - type: epg
+          patterns:
+            - type: tenant
+              operator: contains
+              value: foo
+            - type: ap
+              operator: contains
+              value: bar
+            - type: epg
+              operator: contains
+              value: bar
+    traffic_selector_rules:
+    - ether_type: ip
+      protocol: all
+    - ether_type: arp
+    - ether_type: ip
+      protocol: tcp
+      from_object:
+        source: "1"
+        destination: "2"
+        tcp_flags: ["ack", "fin", "res", "syn" ]
+    state: present
+- name: Delete compliance communication type requirement
+  cisco.nd.nd_compliance_requirement_communication:
+    insights_group: igName
+    name: complianceRequirementName
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.nd.plugins.module_utils.nd import NDModule, nd_argument_spec, sanitize_dict
+from ansible_collections.cisco.nd.plugins.module_utils.ndi import NDI, get_object_selector_payload
+from ansible_collections.cisco.nd.plugins.module_utils.ndi_argument_specs import compliance_base_spec, object_selector_spec, compliance_tcp_spec
+from ansible_collections.cisco.nd.plugins.module_utils.constants import ETHER_TYPES, PROTOCOL_TYPES
+
+
+def main():
+    argument_spec = nd_argument_spec()
+    argument_spec.update(compliance_base_spec())
```

Comment:
Should we use a documentation fragment for this compliance_base_spec part?

Reply:
yes good idea will do